core/templates/pages/preferences-page/preferences-page.component.spec.ts

```diff
@@ -31,63 +26,202 @@ import { TranslatorProviderForTests } from 'tests/test.extras';
-    'by default',
-  function() {
+  it('should save user bio', function() {
+    var userBio = 'User bio example';
+    var isRequestTheExpectOne = function(queryParams) {
+      return decodeURIComponent(queryParams).match('"update_type":"user_bio"');
+    };
+
+    $httpBackend.expect(
+      'PUT', '/preferenceshandler/data', isRequestTheExpectOne).respond(200);
+    ctrl.saveUserBio(userBio);
+    $httpBackend.flush();
+
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should save subject interests after changing it', function() {
+    var subjectInterests = 'Math';
+    var isRequestTheExpectOne = function(queryParams) {
+      return decodeURIComponent(queryParams).match(
+        '"update_type":"subject_interests"');
+    };
+
+    expect(ctrl.subjectInterestsChangedAtLeastOnce).toBe(false);
+
+    $httpBackend.expect(
+      'PUT', '/preferenceshandler/data', isRequestTheExpectOne).respond(200);
+    ctrl.onSubjectInterestsSelectionChange(subjectInterests);
+    $httpBackend.flush();
+
+    expect(ctrl.subjectInterestsChangedAtLeastOnce).toBe(true);
+
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should save preferred site language code after changing it', function() {
+    var newLanguage = 'es';
+    var isRequestTheExpectOne = function(queryParams) {
+      return decodeURIComponent(queryParams).match(
+        '"update_type":"preferred_site_language_code"');
+    };
+
+    $httpBackend.expect(
+      'PUT', '/preferenceshandler/data', isRequestTheExpectOne).respond(200);
+    ctrl.savePreferredSiteLanguageCodes(newLanguage);
+    $httpBackend.flush();
+
+    expect(ctrl.select2DropdownIsShown).toBe(false);
+    $timeout.flush();
+    expect(ctrl.select2DropdownIsShown).toBe(true);
+
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should save preferred audio language code after changing it', function() {
+    var newLanguage = 'es';
+    var isRequestTheExpectOne = function(queryParams) {
+      return decodeURIComponent(queryParams).match(
+        '"update_type":"preferred_audio_language_code"');
+    };
+
+    $httpBackend.expect(
+      'PUT', '/preferenceshandler/data', isRequestTheExpectOne).respond(200);
+    ctrl.savePreferredAudioLanguageCode(newLanguage);
+    $httpBackend.flush();
+
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should show username popover based on its length', function() {
+    expect(ctrl.showUsernamePopover('abcdefghijk')).toBe('mouseenter');
+    expect(ctrl.showUsernamePopover('abc')).toBe('none');
+  });
+
+  it('should save email preferences', function() {
+    var isRequestTheExpectOne = function(queryParams) {
+      return decodeURIComponent(queryParams).match(
+        '"update_type":"email_preferences"');
```

Comment:
```suggestion
        '"update_type": "email_preferences"');
```

Reply:
If adding a new space, the tests will break because it expects exactly this way `"update_type":"email_preferences"` with no space